datafusion/core/tests/dataframe.rs

```diff
@@ -1267,3 +1266,676 @@ async fn use_var_provider() -> Result<()> {
     dataframe.collect().await?;
     Ok(())
 }
+
+// --------------- Begin dataframe function tests ----------------
```

Comment:
these tests are moved from one file to another and `create_test_table` redundancy is removed. Otherwise they are not altered

Reply:
I suggest you put these into rust modules instead. The issue w/ disk space is NOT that they are in different files but that you have two test binaries.

File struct is:

- `dataframe.rs`:
   ```rust
   mod dataframe;
   mod dataframe_functions;
   ```
- `dataframe/dataframe.rs`
- `dataframe/dataframe_functions.rs`

Reply:
This is a good idea -- I implemented a modified version of this suggestion in https://github.com/apache/arrow-datafusion/pull/6697